TestThrottle failed travis

Saw this from Travis-CI on a PR:

```
=== RUN TestThrottle
--- FAIL: TestThrottle (0.05s)
rpcserver_test.go:54: status codes: want: map[200:1 429:1], got: map[200:2]
```

Must be an intermittent issue because this is the first time I've seen this fail since the RPC throttling was added.

--- a/rpc/legacyrpc/rpcserver_test.go
+++ b/rpc/legacyrpc/rpcserver_test.go
@@ -9,15 +9,15 @@ import (
 	"net/http/httptest"
 	"reflect"
 	"testing"
-	"time"
 )
 
 func TestThrottle(t *testing.T) {
 	const threshold = 1
+	busy := make(chan struct{})
 
 	srv := httptest.NewServer(throttledFn(threshold,
 		func(w http.ResponseWriter, r *http.Request) {
-			time.Sleep(20 * time.Millisecond)
+			<-busy
 		}),
 	)
 
@@ -35,6 +35,10 @@ func TestThrottle(t *testing.T) {
 	got := make(map[int]int, cap(codes))
 	for i := 0; i < cap(codes); i++ {
 		got[<-codes]++
+
+		if i == 0 {
+			close(busy)
+		}
 	}
 
 	want := map[int]int{200: 1, 429: 1}

